addons/sale_timesheet/tests/test_reporting.py

```diff
@@ -585,3 +585,109 @@ def test_reversed_downpayment(self):
+        self.assertAlmostEqual(project_stat['expense_amount_untaxed_invoiced'], 0, msg="The expense invoiced amount of the project should be zero, after credit note.")
+        self.assertAlmostEqual(project_stat['expense_cost'], 0, msg="The expense cost of the project should be zero, after credit note.")
+
+    def test_profitability_credit_note_invoice(self):
+        """Test whether the profitability doesn't change with customer invoice or its credit note."""
+        ProjectProfitabilityReport = self.env['project.profitability.report']
+        analytic_account = self.project_global.analytic_account_id
+        product = self.env['product.product'].with_context(mail_notrack=True, mail_create_nolog=True).create({
+            'name': "Product",
+            'standard_price': 100.0,
+            'list_price': 100.0,
+            'taxes_id': False,
+        })
+        test_invoice = self.env['account.move'].create({
+            'type': 'out_invoice',
+            'currency_id': self.env.user.company_id.currency_id,
+            'partner_id': self.partner_customer_usd,
+            'invoice_line_ids': [(0, 0, {
+                'quantity': 1,
+                'product_id': product.id,
+                'price_unit': 100.0,
+                'analytic_account_id': analytic_account.id,
+            })]
+        })
+        test_invoice.action_post()
+        ProjectProfitabilityReport.flush()
+
+        project_stat= ProjectProfitabilityReport.read_group([('project_id', 'in', self.project_global.ids)], ['project_id', 'amount_untaxed_to_invoice', 'amount_untaxed_invoiced', 'timesheet_unit_amount', 'timesheet_cost', 'expense_cost', 'expense_amount_untaxed_to_invoice', 'expense_amount_untaxed_invoiced'], ['project_id'])[0]
+        self.assertAlmostEqual(project_stat['amount_untaxed_invoiced'], 0, msg="The invoiced amount of the project should be zero, before credit note.")
+        self.assertAlmostEqual(project_stat['amount_untaxed_to_invoice'], 0, msg="The amount to invoice of the project should be zero, before credit note.")
+        self.assertAlmostEqual(project_stat['timesheet_unit_amount'], 0, msg="The timesheet unit amount of the project should be zero, before credit note.")
+        self.assertAlmostEqual(project_stat['timesheet_cost'], 0, msg="The timesheet cost of the project should be zero, before credit note.")
+        self.assertAlmostEqual(project_stat['expense_amount_untaxed_to_invoice'], 0, msg="The expense cost to reinvoice of the project should be zero, before credit note.")
+        self.assertAlmostEqual(project_stat['expense_amount_untaxed_invoiced'], 0, msg="The expense invoiced amount of the project should be zero, before credit note.")
+        self.assertAlmostEqual(project_stat['expense_cost'], 0, msg="The expense cost of the project should be equal to the the invoice line price, before credit note.")
```

Comment:
Is this really normal that this is not taken into account in the report?

cc: @tlibioulle

Reply:
No, it's not normal, but it's seems it's not expected to report it in 13.0.

In 14.0 there's an additional field called `other_revenues` which will take it into account, nevertheless, it's not taken into account in 13.0.

The issue this test tries to detect is the fact that neither the invoice or it's credit note are retrieved by the query (before, only the credit note was taken into account, which is obviously ridiculous)

Maybe there's something to check with the PO here.

cc: @auon-odoo

Reply:
Thanks @tlibioulle